cpp/src/copying/scatter.cu

```diff
@@ -164,41 +170,74 @@ struct column_scalar_scatterer_impl<string_view, MapIterator> {
                                      rmm::mr::device_memory_resource* mr,
                                      cudaStream_t stream) const
   {
-    CUDF_FAIL("scatter scalar to list_view not implemented");
+    CUDF_FAIL("scatter scalar to struct_view not implemented");
   }
 };
 
 template <typename MapIterator>
-struct column_scalar_scatterer_impl<struct_view, MapIterator> {
+struct column_scalar_scatterer_impl<dictionary32, MapIterator> {
   std::unique_ptr<column> operator()(std::unique_ptr<scalar> const& source,
                                      MapIterator scatter_iter,
                                      size_type scatter_rows,
                                      column_view const& target,
                                      rmm::mr::device_memory_resource* mr,
                                      cudaStream_t stream) const
   {
-    CUDF_FAIL("scatter scalar to struct_view not implemented");
+    auto dict_target = dictionary::detail::add_keys(
+      dictionary_column_view(target),
+      make_column_from_scalar(*source, 1, rmm::mr::get_current_device_resource(), stream)->view(),
+      mr,
+      stream);
+    auto dict_view    = dictionary_column_view(dict_target->view());
+    auto scalar_index = dictionary::detail::get_index(
+      dict_view, *source, rmm::mr::get_current_device_resource(), stream);
+    auto scalar_iter = indexalator_factory::make_constant_iterator(*scalar_index);
+    auto new_indices = std::make_unique<column>(dict_view.get_indices_annotated(), stream, mr);
+    scalar_iter      = indexalator_factory::make_constant_iterator(*scalar_index);
+    auto target_iter = indexalator_factory::make_output_iterator(new_indices->mutable_view());
+    thrust::scatter(rmm::exec_policy(stream)->on(stream),
+                    scalar_iter,
+                    scalar_iter + scatter_rows,
+                    scatter_iter,
+                    target_iter);
+    // build the dictionary indices column from the result
+    auto const indices_type = new_indices->type();
+    auto const output_size  = new_indices->size();
+    auto const null_count   = new_indices->null_count();
+    auto contents           = new_indices->release();
+    auto indices_column     = std::make_unique<column>(indices_type,
+                                                   static_cast<size_type>(output_size),
+                                                   std::move(*(contents.data.release())),
+                                                   rmm::device_buffer{0, stream, mr},
+                                                   0);
+    // use the keys from the matched column
+    std::unique_ptr<column> keys_column(std::move(dict_target->release().children.back()));
+    // create the output column
+    return make_dictionary_column(std::move(keys_column),
+                                  std::move(indices_column),
+                                  std::move(*(contents.null_mask.release())),
```

Comment:
```suggestion
                                  *(contents.null_mask.release()),
```

Reply:
Removing the `std::move` causes a compile error
```
/copying/scatter.cu -o CMakeFiles/cudf.dir/src/copying/scatter.cu.o
../src/copying/scatter.cu(236): error: no instance of overloaded function "cudf::make_dictionary_column" matches the argument list
            argument types are: (std::unique_ptr<cudf::column, std::default_delete<cudf::column>>, std::unique_ptr<cudf::column, std::default_delete<cudf::column>>, rmm::device_buffer, const cudf::size_type)

```